checkbox-ng/plainbox/impl/highlevel.py

```diff
@@ -116,6 +117,26 @@ def attrs(self):
         """
         return self._attrs
 
+    def find_children_by_name(self, names: list, exact=False):
+        to_explore = [self]
+        name_founds = {name: [] for name in names}
+
+        def match_f(a, b):
+            # a is namespace::id, b may not contain namespace
+            return a == b or a.split("::", 1)[-1] == b
+
+        if exact:
+            match_f = operator.eq
+
+        while to_explore:
+            node = to_explore.pop()
+            matching = [name for name in names if match_f(node.name, name)]
+            for match in matching:
+                name_founds[match].append(node)
+            to_explore += node.children
+
+        return name_founds
```

Comment:
Weird name, I'm not sure if it's correct English, I'd go for `name_matches`
```suggestion
        name_matches = {name: [] for name in names}

        def match_f(a, b):
            # a is namespace::id, b may not contain namespace
            return a == b or a.split("::", 1)[-1] == b

        if exact:
            match_f = operator.eq

        while to_explore:
            node = to_explore.pop()
            matching = [name for name in names if match_f(node.name, name)]
            for match in matching:
                name_matches[match].append(node)
            to_explore += node.children

        return name_matches
```